actor-testkit-typed/src/main/scala/org/apache/pekko/actor/testkit/typed/internal/TestAppender.scala

```diff
@@ -90,6 +91,12 @@ import pekko.annotation.InternalApi
     }
 
     val marker: Option[Marker] = Option(event.getMarkerList).flatMap(_.asScala.headOption)
+    val mdc: Map[String, String] = Option(event.getMDCPropertyMap)
+      .filterNot(_.isEmpty)
+      .orElse(Option(MDC.getMDCAdapter.getCopyOfContextMap))
```

Comment:
if  MDC didn't use`LogbackMDCAdapter` as mdcAdapter, then fallback to the original MDC field.